src/providers/microsoft.rs

```diff
@@ -0,0 +1,23 @@
+use crate::CustomProvider;
+
+pub struct MicrosoftProvider {}
+
+impl MicrosoftProvider {
+    pub fn new(
+        tenant_id: String,
```

Comment:
Could you make this tenant an Enum? I saw in the documentation that we have three types of tenant_id. To avoid the problem of writing an Enum is better.
link of doc: https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-auth-code-flow#request-an-authorization-code

Reply:
There is four different ones, three of them are constant but it can also be an identifier. So either just keep it simple as a string or add en enum with 4 values?

Reply:
You are 100% right, let's keep it as string. tks 👍